sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -301,13 +301,14 @@ abstract class Element<T extends Widget> implements BuildContext {
     // The _updateChild() method returns the new child, if it had to create one,
     // or the child that was passed in, if it just had to update the child, or
     // null, if it removed the child and did not replace it.
-    assert(slot != null);
     if (newWidget == null) {
       if (child != null)
         _detachChild(child);
       return null;
     }
     if (child != null) {
+      if (child._slot != slot)
```

Comment:
shouldn't we only do this if we keep the child? at this point in the code path we still don't know if we're going to throw it away or not.